DistroLauncher-Tests/DistroLauncher-Tests/DistroLauncher-Tests.vcxproj

```diff
@@ -82,8 +82,10 @@ if EXIST "$(SharedIdb)" xcopy /Y /F "$(SharedIdb)" "$(IntDir)"</Command>
     <ClCompile Include="..\..\DistroLauncher\WindowsUserInfo.cpp" />
     <ClCompile Include="..\..\DistroLauncher\WslApiLoader.cpp" />
     <ClCompile Include="..\..\DistroLauncher\WSLInfo.cpp" />
+    <ClCompile Include="ChildProcessTests.cpp" />
     <ClCompile Include="ConsoleServiceTests.cpp" />
     <ClCompile Include="ExtendedCliParserTests.cpp" />
+    <ClCompile Include="FakeChildProcessImpl.cpp" />
```

Comment:
If I understand correctly, you're chosing between real and fake API by deciding what file to link? That seems like it could lead to some hard to debug issues in the future.

I would define a new class in `Testing/mock_api.h`, and define its member functions either where you did or in `Testing/mock_api.cpp`. file and put it in a namespace:
```c++
namespace testing {
    class MockWin32ChildProcess{ /* ... */ };
    using Win32ChildProcess = Win32ChildProcess<MockWin32ChildProcess>;
}

// We can distinguish by the namespace:
Oobe::Win32ChildProcess; // real one
testing::Win32ChildProcess; // mock one
```

Same as with `NamedMutex` and `Sudo`.

Reply:
You're indeed right. I took a bit more aggressive approach making the dependency more visible.